ansible_runner/config/_base.py

```diff
@@ -479,20 +485,17 @@ def wrap_args_for_containerization(self, args, execution_mode, cmdline_args):
                                             dst_mount_path="/runner/artifacts",
                                             labels=":Z")
 
-            # Mount the entire private_data_dir
-            # custom show paths inside private_data_dir do not make sense
-            self._update_volume_mount_paths(new_args,
-                                            "{}".format(self.private_data_dir),
-                                            dst_mount_path="/runner",
-                                            labels=":Z")
         else:
             subdir_path = os.path.join(self.private_data_dir, 'artifacts')
             if not os.path.exists(subdir_path):
                 os.mkdir(subdir_path, 0o700)
 
-            # Mount the entire private_data_dir
-            # custom show paths inside private_data_dir do not make sense
-            self._update_volume_mount_paths(new_args, "{}".format(self.private_data_dir), dst_mount_path="/runner", labels=":Z")
+        # Mount the entire private_data_dir
+        # custom show paths inside private_data_dir do not make sense
+        self._update_volume_mount_paths(new_args, "{}".format(self.private_data_dir), dst_mount_path="/runner", labels=":Z")
+
+        # Mount the stdout callback plugin from the ansible-runner code base
+        self._update_volume_mount_paths(new_args, callback_mount()[0], dst_mount_path=callback_mount()[1], labels=":Z")
```

Comment:
I think this might break if you install runner as root in the system site packages and then run as a non-root user, because you won't be able to relabel a file you dont own.

Reply:
Maybe try `:O` instead of `:Z`?

Reply:
yeah, this sounds more right to me, but I haven't been able to figure out how to artificially create the error to validate it fixes it. I got it to these permissions:

```
bash-4.4$ ls -la /var/lib/awx/venv/awx/lib/python3.8/site-packages/ansible_runner/display_callback/callback/
total 44
drwxr-xr-x. 3 root root  4096 Jan 26 03:18 .
drwxr-xr-x. 3 root root  4096 Jan 26 03:18 ..
-rw-r--r--. 1 root root 30788 Jan 26 03:18 awx_display.py
-rw-r--r--. 1 root root     0 Jan 26 03:18 __init__.py
drwxr-xr-x. 2 root root  4096 Jan 26 03:18 __pycache__
```

Then as the awx user, it still works. Is there any tweak I can do which should make this break?

Reply:
wait, never mind, I was running the wrong thing. I am able to get an error with this type of setup. Right now it just gives the vague message

> ERROR! Invalid callback for stdout specified: awx_display

But I'm still tinkering with the install by root user.

Reply:
Testing the `:0` in one environment, I get

```
Error: mounting overlay failed "/var/lib/awx/venv/awx/lib64/python3.8/site-packages/ansible_runner/display_callback/callback/": chown /var/lib/awx/.local/share/containers/storage/overlay-containers/78dce4e01f5d25bbfe804246e232815dd809cb5360237b1fddf35bc49bfe2fca/userdata/overlay/568435025/upper: invalid argument
```

I'm not sure how big of a problem this is going to be. In another environment it works. Reading related issues, people mention that kernel version needs to be relatively recent for a metacopy setting. Depending on what we support, I could see this getting complicated.

Reply:
Okay, trying on _my own_ machine, I was able to reproduce something by doing only this:

```
sudo chown -R root:root ansible_runner/display_callback/
```

Resulting in

```
$ ansible-runner run demo/ -p test.yml --process-isolation
Error: lsetxattr /home/alancoding/repos/ansible-runner/ansible_runner/display_callback/callback: operation not permitted
```

So maybe I am figuring this out. Here, specifically, with root ownership and `-rw-rw-r--.` permissions, the display callback is not editable for the current user. Like you say, that shouldn't be a problem. I fear that `:O` will not work in all the contexts that we need it to... but it wouldn't be atypical for runner to create a new tmpdir to make something work. We could do this with a copy of the display callback. We know it should only be a few files.

Reply:
Having a replicator, I developed a patch that fixes it.

I did not use the `:O`, because I'm worried this won't work in all cases. Instead, I copied the files (possible if you only have read access) to a tmpdir, and mounted the tmpdir.

Tomorrow I can work on a test case for this logic.